### PR TITLE
Camera pipe fixes and schedule improvements

### PR DESCRIPTION
**Pull request by dsharletg**
This PR changes the camera pipe so scheduling is controlled by the target, rather than a schedule parameter.

It also fixes an underflow issue where random white pixels were generated by unsigned arithmetic in the demosaic wrapping around 0. The fix is to convert the input data to be signed, which matches the C implementation.

### CODE REVIEW

**Comment by steven-johnson** [comment]
LGTM. This should really be converted to use Generator someday.

File: apps/camera_pipe/camera_pipe.cpp [hunk 1]
```diff
@@ -333,13 +332,10 @@ int main(int argc, char **argv) {
         .bound(tx, 0, (out_width/32)*32)
```
Comment (pranavb-ca) [inline apps/camera_pipe/camera_pipe.cpp]: We generate better code if the height and width are a multiple of the tile size. Isn't that what this is trying to do? If so, then even this depends on the schedule which in turn depends on the target.

Comment (dsharletg) [inline apps/camera_pipe/camera_pipe.cpp]: Good point, I'll move it.

**Comment by dsharletg** [comment]
I've reorganized the schedule, please take a look.

**Comment by dsharletg** [comment]
I just noticed another change that I think we should make: the CCM was previously stored as 32 bit integers, I think it should be stored 16 bit. This enables widening 16x16 -> 32 bit multiplies to be used when they are available, and 16 bits should be enough (CCM entries should not exceed 8.8 fixed point).

**Pull request merged**